Source/DataSources/GpxDataSource.js

```diff
@@ -0,0 +1,1223 @@
+            get : function() {
+                return this._clock;
+            }
+        },
+        /**
+         * Gets the collection of {@link Entity} instances.
+         * @memberof GpxDataSource.prototype
+         * @type {EntityCollection}
+         */
+        entities : {
+            get : function() {
+                return this._entityCollection;
+            }
+        },
+        /**
+         * Gets a value indicating if the data source is currently loading data.
+         * @memberof GpxDataSource.prototype
+         * @type {Boolean}
+         */
+        isLoading : {
+            get : function() {
+                return this._isLoading;
+            }
+        },
+        /**
+         * Gets an event that will be raised when the underlying data changes.
+         * @memberof GpxDataSource.prototype
+         * @type {Event}
+         */
+        changedEvent : {
+            get : function() {
+                return this._changed;
+            }
+        },
+        /**
+         * Gets an event that will be raised if an error is encountered during processing.
+         * @memberof GpxDataSource.prototype
+         * @type {Event}
+         */
+        errorEvent : {
+            get : function() {
+                return this._error;
+            }
+        },
+        /**
+         * Gets an event that will be raised when the data source either starts or stops loading.
+         * @memberof GpxDataSource.prototype
+         * @type {Event}
+         */
+        loadingEvent : {
+            get : function() {
+                return this._loading;
+            }
+        }
+    });
+
+    /**
+     * Asynchronously loads the provided GPX data, replacing any existing data.
+     *
+     * @param {String|Document|Blob} data A url, parsed GPX document, or Blob containing binary GPX data or a parsed GPX document.
+     * @param {Object} [options] An object with the following properties:
+     * @param {Number} [options.sourceUri] Overrides the url to use for resolving relative links and other GPX network features.
+     * @returns {Promise} A promise that will resolve to this instances once the GPX is loaded.
+     */
+    GpxDataSource.prototype.load = function(data, options) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(data)) {
+            throw new DeveloperError('data is required.');
+        }
+        //>>includeEnd('debug');
+
+        DataSource.setLoading(this, true);
+
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+        var sourceUri = options.sourceUri;
+
+        var promise = data;
+        if (typeof data === 'string') {
+            promise = loadBlob(proxyUrl(data, this._proxy));
+            sourceUri = defaultValue(sourceUri, data);
+        }
+
+        var that = this;
+        return when(promise, function(dataToLoad) {
+            if (dataToLoad instanceof Blob) {
+                return isZipFile(dataToLoad).then(function(isZip) {
```

Comment:
Unlike KML/KMZ files, pre-zipped gpx files aren't really a thing, so I would remove all code related to unzipping data, it's not needed.